include/pros/optical.h

```diff
@@ -0,0 +1,265 @@
+/**
+ * \file pros/optical.h
+ *
+ * Contains prototypes for functions related to the VEX Optical sensor.
+ *
+ * Visit https://pros.cs.purdue.edu/v5/tutorials/topical/imu.html to learn
+ * more.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2020, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef _PROS_OPTICAL_H_
+#define _PROS_OPTICAL_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#define OPT_GESTURE_ERR (INT8_MAX)
+#define OPT_COUNT_ERR (INT16_MAX)
+#define OPT_TIME_ERR PROS_ERR
+
+#ifdef __cplusplus
+extern "C" {
+namespace pros {
+namespace c {
+#endif
+
+typedef enum optical_direction_e { NO_GESTURE = 0, UP = 1, DOWN = 2, RIGHT = 3, LEFT = 4 } optical_direction_e_t;
+
+typedef struct optical_rgb_s {
+	double red;
+	double green;
+	double blue;
+	double brightness;
+} optical_rgb_s_t;
+
+typedef struct optical_raw_s {
+	uint32_t clear;
+	uint32_t red;
+	uint32_t green;
+	uint32_t blue;
+} optical_raw_s_t;
+
+typedef struct optical_gesture_s {
+	uint8_t udata;
+	uint8_t ddata;
+	uint8_t ldata;
+	uint8_t rdata;
+	uint8_t type;
+	uint8_t pad;
```

Comment:
Do you know if this is actual data or making the implicit padding explicit? If you don't, I wouldn't worry about it, as it's used for the low level "raw" apis. Also, if we have it, some docs on the individual fields would be nice.

Reply:
We don't have docs on the individual fields at this time.